samples/go/main.go

```diff
@@ -3,67 +3,91 @@ package main
-func TWStringCreateWithGoString(s string) unsafe.Pointer {
-    cStr := C.CString(s)
-    defer C.free(unsafe.Pointer(cStr))
-    str := C.TWStringCreateWithUTF8Bytes(cStr)
-    return str
-}
+func main() {
+	fmt.Println("==> calling wallet core from go")
+	str := types.TWStringCreateWithGoString("confirm bleak useless tail chalk destroy horn step bulb genuine attract split")
+	emtpy := types.TWStringCreateWithGoString("")
+	defer C.TWStringDelete(str)
+	defer C.TWStringDelete(emtpy)
 
-// C.TWData -> Go byte[]
-func TWDataGoBytes(d unsafe.Pointer) []byte {
-    cBytes := C.TWDataBytes(d)
-    cSize := C.TWDataSize(d)
-    return C.GoBytes(unsafe.Pointer(cBytes), C.int(cSize))
-}
+	fmt.Println("==> mnemonic is valid: ", C.TWHDWalletIsValid(str))
 
-// Go byte[] -> C.TWData
-func TWDataCreateWithGoBytes(d []byte) unsafe.Pointer {
-    cBytes := C.CBytes(d)
-    defer C.free(unsafe.Pointer(cBytes))
-    data := C.TWDataCreateWithBytes((*C.uchar)(cBytes), C.ulong(len(d)))
-    return data
-}
+	wallet := C.TWHDWalletCreateWithMnemonic(str, emtpy)
+	defer C.TWHDWalletDelete(wallet)
 
-func main() {
-    fmt.Println("==> calling wallet core from go")
-    str := TWStringCreateWithGoString("confirm bleak useless tail chalk destroy horn step bulb genuine attract split")
-    emtpy := TWStringCreateWithGoString("")
-    defer C.TWStringDelete(str)
-    defer C.TWStringDelete(emtpy)
+	key := C.TWHDWalletGetKeyForCoin(wallet, C.TWCoinTypeBitcoin)
+	keyData := C.TWPrivateKeyData(key)
+	defer C.TWDataDelete(keyData)
+
+	fmt.Println("<== bitcoin private key: ", types.TWDataHexString(keyData))
+
+	pubKey, _ := hex.DecodeString("0288be7586c41a0498c1f931a0aaf08c15811ee2651a5fe0fa213167dcaba59ae8")
+	pubKeyData := types.TWDataCreateWithGoBytes(pubKey)
+	defer C.TWDataDelete(pubKeyData)
+	fmt.Println("==> bitcoin public key is valid: ", C.TWPublicKeyIsValid(pubKeyData, C.TWPublicKeyTypeSECP256k1))
+
+	address := C.TWHDWalletGetAddressForCoin(wallet, C.TWCoinTypeBitcoin)
+	defer C.TWStringDelete(address)
+	fmt.Println("<== bitcoin address: ", types.TWStringGoString(address))
+
+	script := C.TWBitcoinScriptLockScriptForAddress(address, C.TWCoinTypeBitcoin)
+	scriptData := C.TWBitcoinScriptData(script)
+	defer C.TWBitcoinScriptDelete(script)
+	defer C.TWDataDelete(scriptData)
+	fmt.Println("<== bitcoin address lock script: ", types.TWDataHexString(scriptData))
+
+	utxoHash, _ := hex.DecodeString("fff7f7881a8099afa6940d42d1e7f6362bec38171ea3edf433541db4e4ad969f")
 
-    fmt.Println("<== mnemonic is valid: ", C.TWHDWalletIsValid(str))
+	utxo := bitcoin.UnspentTransaction{
+		OutPoint: &bitcoin.OutPoint{
+			Hash:     utxoHash,
+			Index:    0,
+			Sequence: 4294967295,
+		},
+		Amount: 625000000,
+		Script: types.TWDataGoBytes(scriptData),
+	}
 
-    wallet := C.TWHDWalletCreateWithMnemonic(str, emtpy)
-    defer C.TWHDWalletDelete(wallet)
+	plan := bitcoin.TransactionPlan{
+		Amount: 625000000,
+		Utxos:  []*bitcoin.UnspentTransaction{&utxo},
+	}
```

Comment:
Plan should either be left empty, or created by an invocation to plan (AnyPlan).  I suggest to leave it empty.

Reply:
done